ovs-service/src/main/java/org/dcsa/ovs/service/VesselScheduleService.java

```
@@ -1,21 +1,172 @@
-  private final VesselScheduleRepository repository;
-  private final VesselScheduleMapper mapper;
 
-  public List<VesselScheduleTO> findAll() {
-    return ListMapper.convertList(repository.findAll(), mapper::toTO);
+  private final ServiceRepository serviceRepository;
+
+  @Builder
+  public static class ServiceSchedulesFilters {
+    String carrierServiceCode;
+    String universalServiceReference;
+    String vesselIMONumber;
+    String vesselName;
+    String voyageNumber;
+    String universalVoyageReference;
+    String unLocationCode;
+    String facilitySMDGCode;
+    String startDate;
+    String endDate;
+    Integer limit;
+    String cursor;
+    String apiVersion;
+  }
+
+  public List<VesselScheduleTO> findAll(ServiceSchedulesFilters requestFilters) {
+
+    return serviceRepository.findAll().stream()
+        .map(
+            service ->
+                VesselScheduleTO.builder()
+                    .carrierServiceName(service.getCarrierServiceName())
+                    .carrierServiceCode(service.getCarrierServiceCode())
+                    .universalServiceReference(service.getUniversalServiceReference())
+                    .vessels(extractVesselsToVesselTO(service.getVessels()))
+                    .build())
+        .toList();
+  }
+
+  private List<VesselTO> extractVesselsToVesselTO(List<Vessel> vessels) {
+    return vessels.stream()
+        .map(
+            vessel ->
+                VesselTO.builder()
+                    .vesselName(vessel.getName())
+                    .vesselIMONumber(vessel.getImoNumber())
+                    .vesselOperatorCarrierSMDGCode(vessel.getOperatorCarrier().getSmdgCode())
+                    .vesselCallSign(vessel.getCallSign())
+                    .vesselLength(vessel.getLength())
+                    .vesselWidth(vessel.getWidth())
+                    .dimensionUnit(
+                        DimensionUnit.valueFromString(vessel.getDimensionUnit()).orElse(null))
```

If the value can't be mapped (and is not null) wouldn't something be horrible wrong and we should throw an exception?

---

I would favor either an exception or a change to our API where we set a default dimensionUnit

---

For now,  As discussed with @pimg it will be left as is.